test/unittests/common_query/test_common_query.py

```diff
@@ -72,34 +77,37 @@ def test_common_query_events(self):
             # stop thinking animation
             {'type': 'enclosure.mouth.reset',
              'data': {},
-             'context': {'destination': "audio", 'source': 'skills',
-                         'skill_id': "wiki.test"}
-             },
-            # skill callback event
-            {'type': 'question:action',
-             'data': {'skill_id': 'wiki.test',
-                      'phrase': 'what is the speed of light',
-                      'callback_data': {'query': 'what is the speed of light',
-                                        'answer': 'answer 1'}},
-             'context': skill_ans_ctxt},
-            # tell enclosure about active skill (speak method)
+             'context': qq_ctxt},
+            # tell enclosure about active skill (speak method). This is the
+            # skill that provided the response and may follow-up with actions
+            # in a callback method
             {'type': 'enclosure.active_skill',
              'data': {'skill_id': 'wiki.test'},
-             'context': {'destination': "audio", 'source': 'skills',
-                         'skill_id': 'wiki.test'}},
-            # execution of speak method
+             'context': qq_ctxt},
+            # execution of speak method. This is called from CommonQuery, but
```

Comment:
NOTE: skill is supposed to do the speak, CommonQuery doing it is a backwards compat thing from classic mycroft

speak is usually done by skills here https://github.com/OpenVoiceOS/OVOS-workshop/blob/dev/ovos_workshop/skills/common_query_skill.py#L252

Reply:
Updated to handle the `handles_speech` data and added/updated tests for both cases